config/initializers/rack_attack.rb

```diff
@@ -0,0 +1,40 @@
+class Rack::Attack
+  # cache store
+  Rack::Attack.cache.store = ActiveSupport::Cache::MemoryStore.new
+
+  # whitelist NU network
+  safelist_ip("165.124.0.0/16")
+  safelist_ip("129.105.0.0/16")
+
+  # Always allow requests from localhost
+  # (blocklist & throttles are skipped)
+  Rack::Attack.safelist('allow from localhost') do |req|
+    '127.0.0.1' == req.ip || '::1' == req.ip
+  end
+
+  # These include some requests by the application, and not by the user
+  Rack::Attack.safelist('ignorable paths') do |req|
+    req.fullpath.start_with?('/assets',
+                             '/users/notifications_number', 
+                             '/catalog/facet/tags_sim.json',
+                             '/downloads/')
+  end
+
+  catalog_limit_proc = proc { |req| req.env['warden'].user.blank? ? 10 : 500 }
+  # Throttle /catalog by subnet
+  # 500/6mins for authenticated, 10/6mins for unauthenticated
+  # Key: "rack::attack:#{Time.now.to_i/:period}:catalog/subnet:#{subnet of req.ip}"
+  throttle('catalog/subnet', limit: catalog_limit_proc, period: 6.minutes) do |req|
+    if req.fullpath.start_with?('/catalog?', '/catalog/') && !req.fullpath.start_with?('/catalog?page')
+      req.ip.slice(0..req.ip.rindex("."))
```

Comment:
Let's document in comments that we are throttling every request based on their **/24** subnet. It will be interesting to see how users react to this in practice; throttling by subnet like this may result in situations where a user is poisoning the well for everyone else. (which is why this should not be done for an API)

Reply:
`poisoning the well` - great analogy.